packages/@lwc/template-compiler/src/codegen/helpers.ts

```diff
@@ -6,19 +6,113 @@
  */
 import * as t from '../shared/estree';
 import { toPropertyName } from '../shared/utils';
-import { BaseElement, ChildNode, LWCDirectiveRenderMode, Node } from '../shared/types';
+import {
+    Attribute,
+    BaseElement,
+    ChildNode,
+    Element,
+    LWCDirectiveRenderMode,
+    Node,
+    Property,
+} from '../shared/types';
 import {
     isParentNode,
     isSlot,
     isForBlock,
     isBaseElement,
     isIf,
     isDynamicDirective,
+    isElement,
+    isText,
+    isComment,
 } from '../shared/ast';
 import { TEMPLATE_FUNCTION_NAME, TEMPLATE_PARAMS } from '../shared/constants';
 
 import CodeGen from './codegen';
 
+function serializeAttrs(element: Element): string {
+    /**
+     * 0: styleToken in existing class attr
+     * 1: styleToken for added class attr
+     * 2: styleToken as attr
+     */
+    const attrs: string[] = [];
+    let hasClassAttr = false;
+
+    const collector = ({ name, value }: Attribute | Property) => {
+        let v = (value as any).value;
+
+        if (name === 'class') {
+            hasClassAttr = true;
+            v += '${0}';
+        }
+        if (typeof v === 'string') {
+            attrs.push(`${name}="${v}"`);
+        } else {
+            attrs.push(name);
+        }
+    };
+
+    element.attributes.forEach(collector);
+    element.properties.forEach(collector);
+
+    return attrs.join(' ') + (hasClassAttr ? '${2}' : '${1}${2}');
+}
+
+function serializeChildren(children: ChildNode[]): string {
+    let html = '';
+
+    children.forEach((child) => {
+        if (isElement(child)) {
+            html += serializeStaticElement(child);
+        } else if (isText(child)) {
+            html += (child.value as any).value;
+        } else if (isComment(child)) {
+            html += `<!--${child.value}-->`;
```

Comment:
This also needs to be escaped, otherwise it's also vulnerable:

```html
    <h1>
        <!-- --&gt;&lt;script&gt;alert('pwned')&lt;/script&gt; &lt!-- -->
    </h1>
```

![Screen Shot 2022-04-12 at 1 43 39 PM](https://user-images.githubusercontent.com/283842/163050391-f46f398c-c884-4679-a9a2-d1549e9680d8.png)

Reply:
resolved in https://github.com/salesforce/lwc/pull/2781/commits/582ed58052feb00e87fed4a7587b59e044687e83